packages/compile-solidity/src/compilerSupplier/loadingStrategies/VersionRange.ts

```diff
@@ -1,20 +1,82 @@
-const debug = require("debug")("compile:compilerSupplier");
-const requireFromString = require("require-from-string");
-const fs = require("fs");
-const originalRequire = require("original-require");
-const axios = require("axios").default;
-const semver = require("semver");
-const solcWrap = require("solc/wrapper");
-const LoadingStrategy = require("./LoadingStrategy");
-
-class VersionRange extends LoadingStrategy {
+import debugModule from "debug";
+const debug = debugModule("compile:compilerSupplier");
+
+import requireFromString from "require-from-string";
+import originalRequire from "original-require";
+import axios from "axios";
+import semver from "semver";
+import solcWrap from "solc/wrapper";
+import { Cache } from "../Cache";
+import { observeListeners } from "../observeListeners";
+import { NoVersionError, NoRequestError } from "../errors";
+
+export class VersionRange {
```

Comment:
@haltman-at notes [here](https://github.com/trufflesuite/truffle/pull/4222#pullrequestreview-728040585) that we may be able to make use of the semver package's `maxSatisfying` function, eliminating this class's `findNewestValidVersion` method.

Reply:
Looking to make this change, these behaviors are rather different - `maxSatisfying` takes a list of version, but our `findNewestValidVersion` takes this structured data of `{ releases, prereleases, builds }` that is nontrivial to dissect and clean up.

I posit that it's not worth the risk + effort to do this right now.

Reply:
The only information from that structure that it uses, however, is `Object.keys(releases)`; it discards the rest.  So I don't think it's so far off as you're suggesting.

Reply:
Alright, I switched to using `maxSatisfying`, but it wasn't clean enough to warrant inlining, so I've kept `findNewestValidVersion`

Reply:
OK, makes sense.